src/ORM/Query.php

```diff
@@ -1131,8 +1132,13 @@ protected function _execute(): ResultSetInterface
             return new $decorator($this->_results);
         }
 
-        $statement = $this->getEagerLoader()->loadExternal($this, $this->execute());
-        $resultset = new ResultSet($this, $statement->fetchAll('assoc') ?: []);
+        $results = $this->execute()->fetchAll(StatementInterface::FETCH_TYPE_ASSOC);
+        if ($results === false) {
```

Comment:
why checking for `false` ?

Reply:
php8 changed the return type of fetchAll() to always be an array. Don't know if you made this change already.

Reply:
The return type of `Statement::fetchAll()` is still `array|false`. So the check for `false` is needed for now to keep static analyzers happy.